src/extension/src/azure/azure-resource-group/resourceGroupModule.ts

```diff
@@ -0,0 +1,96 @@
+import { SubscriptionItem } from "../azure-auth/azureAuth";
+import { ResourceManager } from "../azure-arm/resourceManager";
+import { ResourceGroup } from "azure-arm-resource/lib/resource/models";
+import { ResourceManagementClient } from "azure-arm-resource/lib/resource/resourceManagementClient";
+import { ResourceGroupError, AuthorizationError } from "../../errors";
+import { CONSTANTS } from "../../constants";
+
+const VALIDATION_LIMIT = 3;
+
+export interface ResourceGroupSelection {
+    subscriptionItem: SubscriptionItem;
+    resourceGroupName: string;
+    location: string;
+}
+
+export class ResourceGroupDeploy {
+    private azureResourceClient: ResourceManagementClient | undefined;
+
+    private setAzureResourceClient (userSubscriptionItem: SubscriptionItem) {
+        if (
+            this.azureResourceClient === undefined ||
+            this.azureResourceClient.subscriptionId !== userSubscriptionItem.subscriptionId
+            ) {
+                this.azureResourceClient = new ResourceManager().getResourceManagementClient(userSubscriptionItem);
+            }
+    }
+
+    public async createResourceGroup(
+        resourceGroupSelection: ResourceGroupSelection
+    ): Promise<string | undefined> {
+        const {subscriptionItem, resourceGroupName, location} = resourceGroupSelection;
+
+        let azureResourceClient: ResourceManagementClient = new ResourceManager().getResourceManagementClient(subscriptionItem);
+
+        let options: ResourceGroup = {
```

Comment:
When we don't modify a variable after first assignment, we should use const instead of let.